src/cmap/errors.ts

```diff
@@ -28,9 +27,9 @@ export class WaitQueueTimeoutError extends MongoDriverError {
   /** The address of the connection pool */
   address: string;
 
-  constructor(pool: Connection | ConnectionPool) {
-    super('Timed out while checking out a connection from connection pool');
-    this.address = pool.address;
+  constructor(message: string, address: string) {
+    super(`Timed out while checking out a connection from connection pool${message}`);
```

Comment:
I think from a human reading perspective, it would make more sense for the metrics function to return the metrics formatted as a standalone sentence, and for this function to assume that the message will be a standalone sentence... just because it feels a little odd to have a constructor that takes a seemingly arbitrary message assume that it can always be appended directly to `pool`.

Reply:
Moved the full error message into the metrics object itself, and the complete message is always passed in and used as is.